app/queries/get_change_offer_options.rb

```diff
@@ -0,0 +1,84 @@
+class GetChangeOfferOptions
+  attr_accessor :user, :for_provider, :recruitment_cycle_year
+
+  # application_choice.offered_course (either current or the original)
+  def initialize(user:, current_course:)
+    @user = user
+    @for_provider = current_course.accredited_provider || current_course.provider
```

Comment:
```suggestion
    @provider = current_course.accredited_provider || current_course.provider
```
I think `for_provider` is a bit confusing. I would have expected that to be used in an action if anywhere, but as a variable I don't see any readability benefits.

Reply:
Would `@accredited_provider` be better? We use this to select courses ratified by this provider, which, for self-ratified courses, is the training provider. So just `@provider` seems simple/deceptive to me.

Reply:
Yes, `@accredited_provider` works too!